src/transformers/models/beit3/modeling_beit3.py

```diff
@@ -0,0 +1,1447 @@
+            loss=loss,
+            logits=logits,
+            hidden_states=outputs.hidden_states,
+            attentions=outputs.attentions,
+        )
+
+
+def contrastive_loss(logits: torch.Tensor) -> torch.Tensor:
+    return nn.functional.cross_entropy(logits, torch.arange(len(logits), device=logits.device))
+
+
+# Copied from transformers.models.clip.modeling_clip.clip_loss
+def clip_loss(similarity: torch.Tensor) -> torch.Tensor:
+    caption_loss = contrastive_loss(similarity)
+    image_loss = contrastive_loss(similarity.t())
+    return (caption_loss + image_loss) / 2.0
+
+
+@add_start_docstrings(
+    """BEiT-3 Transformer model with language and image classifier heads on top for image-text retrieval.""",
+    BEIT3_START_DOCSTRING,
+)
+class Beit3ForImageTextRetrieval(Beit3PreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+        embed_dim = config.hidden_size
+        self.beit3 = Beit3Model(config)
+        self.language_classifier = nn.Linear(embed_dim, embed_dim, bias=False)
+        self.vision_classifier = nn.Linear(embed_dim, embed_dim, bias=False)
+        self.logit_scale = nn.Parameter(torch.ones([]) * config.logit_scale_init_value)
+        self.post_init()
+
+    @add_start_docstrings_to_model_forward(BEIT3_FOR_TEXT_RETRIEVAL_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=Beit3ImageTextMatchingOutput, config_class=_CONFIG_FOR_DOC)
+    def forward(
+            self,
+            input_ids: torch.LongTensor,
+            pixel_values: torch.FloatTensor,
+            attention_mask: Optional[torch.Tensor] = None,
+            return_loss: Optional[bool] = None,
+            output_hidden_states: Optional[bool] = None,
+            output_attentions: Optional[bool] = None,
+            return_dict: Optional[bool] = None,
+    ) -> Union[Tuple[Any], Beit3ImageTextMatchingOutput]:
+        r"""
+        Returns:
+
+        Examples:
+
+        ```python
+        >>> from transformers import Beit3Processor, Beit3ForImageTextRetrieval
+        >>> from PIL import Image
+        >>> import requests
+
+        >>> processor = Beit3Processor.from_pretrained("Raghavan/beit3_base_patch16_384_coco_retrieval")
+        >>> model = Beit3ForImageTextRetrieval.from_pretrained("Raghavan/beit3_base_patch16_384_coco_retrieval")
+
+        >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        >>> image = Image.open(requests.get(url, stream=True).raw)
+        >>> text = "This is photo of a cat"
+
+        >>> inputs = processor(text=text, images=image)
+        >>> another_input_ids = processor(text=["This is photo of a dog"], images=image, return_attention_mask=False)
+        >>> # forward pass
+        >>> outputs = model(input_ids=torch.tensor([inputs["input_ids"][0], another_input_ids["input_ids"][0]]),
+        ...    pixel_values=torch.tensor([inputs["pixel_values"][0], inputs["pixel_values"][0]]),
+        ...    return_loss=True)
```

Comment:
This example is a bit weird. First of all, inputs are always prepared by specifying `return_tensors="pt"` to the processor. So there's no need to use `torch.tensor` yourself. Also specifying `return_attention_mask=False` is weird and should not be the case.

Reply:
+1